tools/data_loader.py

```diff
@@ -0,0 +1,378 @@
+    else:
+        src =cfg['data_load'][data_source]
+
+    with status(f"Connecting to database {cfg['database']['database']}"):
+        init_db(**cfg["database"])
+
+    if src.get("drop_tables", False):
+        with status("Dropping all tables"):
+            drop_tables()
+
+    if src.get("create_tables", False):
+        with status("Creating tables"):
+            create_tables()
+
+    if src.get("print_tables", False):
+        for model in Base.metadata.tables:
+            print("    -", model)
+
+    if src.get("create_permissions", False):
+        with status(f"Creating permissions"):
+            setup_permissions()
+
+    if src.get("users") is not None:
+        with status(f"Creating users"):
+
+            users = []
+            for u in src.get('users', []):
+                users.append(
+                    User(username=u["username"], role_ids=u["role_ids"])
+                )
+            DBSession().add_all(users)
+            for u in users:
+                DBSession().add(
+                    TornadoStorage.user.create_social_auth(u, u.username, "google-oauth2")
+                )
+            users_dict = {u.username: u.id for u in users}
+
+        with status("Creating tokens"):
+
+            tokens = {}
+            for u in src.get('users', []):
+                for t in u.get("tokens", []):
+                    tokens[t["name"]] = create_token(
+                        t["permissions"],
+                        users_dict[u["username"]],
+                        t["name"]
+                    )
+                    if t.get("print", False):
+                        print(t["name"], tokens[t["name"]])
+
+    def assert_post(endpoint, data, token):
+        response_status, data = api("POST", endpoint, data, token)
+        if not response_status == 200 and data["status"] == "success":
+            raise RuntimeError(
+                f'API call to {endpoint} failed with status {status}: {data["message"]}'
+            )
+        return data
+
+    with status("Launching web app & executing API calls"):
+        try:
+            response_status, data = api("GET", "sysinfo", token=list(tokens.values())[0])
+            app_already_running = True
+        except requests.ConnectionError:
+            app_already_running = False
+            web_client = subprocess.Popen(
+                ["make", "run"], cwd=basedir, preexec_fn=os.setsid
+            )
+
+        server_url = f"http://localhost:{cfg['ports.app']}"
+        print()
+        print(f"Waiting for server to appear at {server_url}...")
+
+        try:
+            verify_server_availability(server_url)
+            print("App running - continuing with API calls")
+
+            if src.get("groups") is not None:
+                with status("Creating groups"):
+                    group_ids = []
+                    group_dict = {}
+                    filter_dict = {}
+                    for g in src.get('groups', []):
```

Comment:
The preponderance of single-character variable names makes this file much harder to read for me. I'd request more descriptive variable names (e.g. `group` instead of `g`) throughout this file. Otherwise, when jumping in to edit/debug, one has to hunt down the assignment statement to have any idea of what the variable is, whereas longer, more descriptive names are pretty obvious and easier to read.

Reply:
General comment (not a request from @profjsb) -- for refactoring PyCharm is great, it constructs an abstract syntax tree for your code and identifies all the uses of a variable throughout the module, then lets you rename them all at once